### DISCUSSION TITLE
Pocketbase Transactions NextJS 14 with Javascript

### DISCUSSION THREAD

**Discussion by Nils19024** [pocketbase/pocketbase#4683] [Q&A] [answered]
I am using NextJS 14 (Javascript) with Pocketbase. How can I use the transaction function provided in the "Extend with Javascript" documentation:

```
$app.dao().runInTransaction((txDao) => {
    // update a record
    const record = txDao.findRecordById("articles", "RECORD_ID")
    record.set("status", "active")
    txDao.saveRecord(record)
    // run some custom raw query
    txDao.db().newQuery("DELETE FROM articles WHERE status = 'pending'").execute()
})
```

Where does the $app variable come frome? I have a server action where i access the database, which looks like this:

```
"use server"

export async function remittance(memberNFC, money) {
  try {
    const cookie = cookies().get('pb_auth');
    pb.authStore.loadFromCookie(cookie.value);
    
    // Here i wanted to add the database transaction
    await pb.collection('users' + pb.authStore.model.club).delete(id);
    await pb.collection('members_' + pb.authStore.model.club).delete(id);
    // Here i wanted to add the database transaction

    return true
  } catch (error) {
    console.log(error)
    return false;
  }
}
```

How can i implement it into my code. I already tried `pb.dao().runInTransaction((txDao) => {})`, but it says: 

> pb.dao() is not a function.

Can anyone help me with this problem and explain how it works and how i can implemented into my code? Thanks :)

**Comment by ganigeorgiev** [comment, marked as answer]
I don't understand what you are trying todo but the PocketBase JSVM (goja) can be used only as part of the `pb_hooks` and `pb_migrations`, aka. it is not a node runtime and cannot be used inside your Next.js code. For more details please read https://pocketbase.io/docs/js-overview/.

**Reply by Nils19024** [reply]
ok thank you

**Reply by Nils19024** [reply]
one more question. How can i implement a transaction in nextjs with pocketbase by myself? If one database operation will fail, all the other operations should stop and should not be commited to the database.

**Reply by ganigeorgiev** [reply]
There is no support for transactional/bulk operations over the Web APIs at the moment, if that's what you are asking.

Transactions in PocketBase can be executed only if you are [using PocketBase as framework](https://pocketbase.io/docs/use-as-framework/), eg. by registering a custom PocketBase route (with [Go](https://pocketbase.io/docs/go-routing/) or [JS pb_hooks](https://pocketbase.io/docs/js-routing/)).

**Reply by Nils19024** [reply]
Yes that is what i wanted, but okay i will try to implement a similiar logic by myself. Thank you 👍
